### PR TITLE
Chore: Convert the consumer to a plugin

### PR DESCRIPTION
**Pull request by stumpylog**
## Proposed change

Followup to the other changes, this took a lot more time, since so many tests relied on the particular method.

Draft for a few reasons, to check the coverage, check the result, make sure I'm happy with the changes

Related to: #5367

## Type of change

- [ ] Bug fix: non-breaking change which fixes an issue.
- [ ] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

- [ ] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [ ] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [ ] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [ ] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [ ] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [ ] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by shamoon** [comment]
Ha, yea this one is a big diff! Let me know if you'd like any testing beyond just playing around with it. I already have and didnt notice any issues with basic stuff

**Review by shamoon — APPROVED** [review]
Again, nothing unexpected I saw here, otherwise relying on the test suite here re code changes. Thanks for all the work on this project

Screenshot 2024-04-12 at 12 22 19 AM

File: paperless-ngx.code-workspace
Comment (shamoon) [inline paperless-ngx.code-workspace:1]: Ps. Love this, thanks

**Comment by stumpylog** [comment]
I'm not sure what line it thinks is missing.  I think I'll just accept the super slight dip.

**Pull request merged**